boot/spring_cloud_bindings.go

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package boot
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/buildpacks/libcnb"
+	"github.com/paketo-buildpacks/libpak"
+	"github.com/paketo-buildpacks/libpak/bard"
+	"github.com/paketo-buildpacks/libpak/sherpa"
+
+	_ "github.com/paketo-buildpacks/spring-boot/boot/statik"
+)
+
+var (
+	bindingsProfileScript = "enable-bindings.sh"
+)
+
+type SpringCloudBindings struct {
+	ApplicationPath  string
+	LayerContributor libpak.DependencyLayerContributor
+	Logger           bard.Logger
+}
+
+func NewSpringCloudBindings(
+	applicationPath string,
+	dependency libpak.BuildpackDependency,
+	cache libpak.DependencyCache,
+	plan *libcnb.BuildpackPlan) SpringCloudBindings {
+	return SpringCloudBindings{
+		ApplicationPath: applicationPath,
+		LayerContributor: libpak.NewDependencyLayerContributor(dependency, cache, plan),
+	}
+
+}
+
+//go:generate statik -src . -include *.sh
+func (s SpringCloudBindings) Contribute(layer libcnb.Layer) (libcnb.Layer, error) {
+	s.LayerContributor.Logger = s.Logger
+
+	jarName := filepath.Base(s.LayerContributor.Dependency.URI)
+	jarPath := filepath.Join(layer.Path, jarName)
+	bootDeps := filepath.Join(s.ApplicationPath, "BOOT-INF", "lib")
+	if err := os.MkdirAll(bootDeps, 0777); err != nil {
+		return libcnb.Layer{}, fmt.Errorf("unable to ensure BOOT-INF/lib exists\n%w", err)
+	}
+	if err := os.Symlink(jarPath, filepath.Join(bootDeps, jarName)); err != nil {
+		return libcnb.Layer{}, fmt.Errorf("unable to link spring cloud bindings into BOOT-INF\n%w", err)
+	}
+
+	return s.LayerContributor.Contribute(layer, func(artifact *os.File) (libcnb.Layer, error) {
```

Comment:
Move `LayerContributor` [above the every-time functionality](https://github.com/paketo-buildpacks/libbs/blob/main/application.go#L84-L113).